### DISCUSSION TITLE
recover

### DISCUSSION THREAD

**Discussion by heinono1** [idaholab/moose#22656] [Q&A General] [answered]
If I use --recover (checkpoints and split mesh), how does the recover read the input file? Specifically, I read a file using PiecewiseMultilinear in the input file. If I change the file that is read by PiecewiseMultilinear, will the values from the new file be read in on the recover, or are the old values stored and be used? What this really getting to is trying to use a previously equilibrated run (using split-mesh) as a starting point for a new run, but with changed driving function read by PiecewiseMultilinear.
It will be awesome to get a restart capability with using Nemesis split files as starting point....

**Comment by GiudGiud** [comment, marked as answer]
Hello

The input file is not saved to the recover checkpoint. It is read again. The checkpoint provides solution vectors, which if the input file is compatible with, will be used to recover the solution. 
Recover is really meant to start the same exact simulation, except it crashed for X reason the first time.

I think you may want to look into checkpoint restart, by specifying the restart_file_base in the Problem. 

You may change the data file from PiecewiseMultilinear when performing any kind of restart or recover. 

Guillaume

**Reply by heinono1** [reply]
Thanks, Guillaume!
